encoder.go

```diff
@@ -85,9 +91,12 @@ func (encoder *encoder) Encode(data any) (wo *bindings.WafObject, err error) {
 	value := reflect.ValueOf(data)
 	wo = &bindings.WafObject{}
 
-	err = encoder.encode(value, wo, encoder.objectMaxDepth)
-	if len(encoder.truncations[ObjectTooDeep]) != 0 {
-		encoder.measureObjectDepth(value, time.Millisecond)
+	stopwatch.TimedByStopwatch(encoder.watch, func() {
+		err = encoder.encode(value, wo, encoder.objectMaxDepth)
+	})
+
+	if len(encoder.truncations[ObjectTooDeep]) != 0 && !encoder.watch.Expired() {
+		encoder.measureObjectDepth(value, encoder.watch.Remaining())
 	}
```

Comment:
MAYBE we wanna do this even if the watch expired? (but also maybe not)

The `measureObjectDepth` function caps itself to a 1ms time budget.